lib/companies.js

```diff
@@ -1,57 +1,75 @@
 'use strict'
 
+const isFunction = require('101/is-function')
 const Promise = require('bluebird')
+const Util = require('./util')
 
 /**
  * Companies - Used for communication with intercomClient's company object
  * @class
  * @author Ryan Kahn
  */
 class Companies {
-
   /**
    * Constructor for Companies
-   * @param {Orion} orion - Instance of Orion which is used for the intercom client and canUseIntercom methods
+   * @param {object} client Intercom client for customer related actions.
    */
-  constructor (orion) {
-    this.orion = orion
+  constructor (client) {
+    this.client = client
   }
 
   /**
-   * Helper method to wrap all the intercomClient.companies calls around canUseIntercom
-   * @param {String} method
-   * @returns {Promise} - Returns success if intercom is disabled,
-   * otherwise returns results of call to intercom client
+   * Helper method to wrap all the intercomClient.companies calls around
+   * `canUseIntercom`
+   * @param {String} method Name of the method to wrap for the intercom
+   *   companies client.
+   * @returns {Promise} - Returns success if intercom is disabled, otherwise
+   *   returns results of call to intercom client
    * @private
    */
   _wrap (method) {
```

Comment:
Long term - We are probably going to want this in a base class that's inherited.